opencog/atomese/interpreter/condlink_interpreter.h

```diff
@@ -0,0 +1,82 @@
+/** condlink_interpreter.h ---
+ *
+ * Copyright (C) 2019 OpenCog Foundation
+ *
+ * Authors: Kiros Hailay <kiros.hailay12@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License v3 as
+ * published by the Free Software Foundation and including the exceptions
+ * at http://opencog.org/wiki/Licenses
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program; if not, write to:
+ * Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#ifndef ASMOSES_CONDLINK_INTERPRETER_H
+#define ASMOSES_CONDLINK_INTERPRETER_H
+
+#include <opencog/atoms/base/Handle.h>
+#include <opencog/atoms/value/LinkValue.h>
+#include <opencog/utils/valueUtils.h>
+#include <boost/iterator/zip_iterator.hpp>
+
+namespace opencog { namespace atomese {
+
+/**
+ * this struct is expected to be used in boost::zip_iterator to perform
+ * condlink on LinkValuePtr
+ * and it has method operator overloading that takes three ValuePtr which are the conditions, experssions, and defaults
+ *
+*/
+struct zip_cond :
+			public std::unary_function<const boost::tuple
+					<const ValuePtr &, const ValuePtr &, const ValuePtr &> &, void> {
+		ValueSeq _result;
```

Comment:
This formatting is weird too, ideally you'd use smart tabs (also called mixed tabs), meaning tabs for indentation, space for alignment.